modules/siddhi-extensions/event-table/src/test/java/org/wso2/siddhi/extension/eventtable/hazelcast/DefineTableTestCase.java

```diff
@@ -0,0 +1,471 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.eventtable.hazelcast;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.Member;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+import org.wso2.siddhi.core.ExecutionPlanRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.query.api.ExecutionPlan;
+import org.wso2.siddhi.query.api.annotation.Annotation;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+import org.wso2.siddhi.query.api.definition.TableDefinition;
+import org.wso2.siddhi.query.api.exception.DuplicateDefinitionException;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+import org.wso2.siddhi.query.api.execution.query.Query;
+import org.wso2.siddhi.query.api.execution.query.input.stream.InputStream;
+import org.wso2.siddhi.query.compiler.exception.SiddhiParserException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class DefineTableTestCase {
+    static final Logger log = Logger.getLogger(DefineTableTestCase.class);
+    private static final long RESULT_WAIT = 500;
+
+    @Test
+    public void testQuery1() throws InterruptedException {
+        log.info("testTableDefinition1 - OUT 0");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        TableDefinition tableDefinition = TableDefinition.id("cseEventStream").annotation(Annotation.annotation("from").element("eventtable", "hazelcast")).attribute("symbol", Attribute.Type.STRING).attribute("price", Attribute.Type.INT);
+        ExecutionPlan executionPlan = new ExecutionPlan("ep1");
+        executionPlan.defineTable(tableDefinition);
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(executionPlan);
+        List<String> hciNames = new ArrayList<String>();
+        for (HazelcastInstance hci : Hazelcast.getAllHazelcastInstances()) {
+            hciNames.add(hci.getName());
+        }
+        Assert.assertTrue(hciNames.contains(HazelcastEventTableConstants.HAZELCAST_INSTANCE_PREFIX + executionPlanRuntime.getName()));
+        executionPlanRuntime.shutdown();
```

Comment:
@grainier Better call the ExecutionPlanRuntime.shutdown() method within a finally block as a best practice so that assertion failures do not cause leaking of resources.